app/services/distribution_itemized_breakdown_service.rb

```diff
@@ -0,0 +1,81 @@
+class DistributionItemizedBreakdownService
+  #
+  # Initializes the DistributionItemizedBreakdownService whoms
+  # purpose to construct a itemized breakdown of items distributed
+  # and what is left on-hand currently (at the time of running)
+  #
+  # @param organization [Organization]
+  # @param distribution_ids [Array<Integer>]
+  # @return [DistributionItemizedBreakdownService]
+  def initialize(organization:, distribution_ids:)
+    @organization = organization
+    @distributions = organization.distributions.where(id: distribution_ids).includes(line_items: :item)
```

Comment:
Looking at this again, it seems a bit odd that an `initialize` method does a database query. Wouldn't this fit better inside `fetch`?

Reply:
Good point. Yeh it is better to put the call to query inside the `fetch`.